src/dnx/dnx.cpp

```diff
@@ -391,6 +391,14 @@ int wmain(int argc, wchar_t* argv[])
                 }
 
                 ::wprintf_s(L"Debugger attached.\r\n");
+
+                //remove --debug flag from arguments
+                if (argc > i)
```

Comment:
Should be argc > i + 1
Also, need to move the argv[i] = L"\0"; to outside of the if check